lib/serviceClient.ts

```diff
@@ -551,7 +551,13 @@ export function flattenResponse(_response: HttpOperationResponse, responseSpec:
     const modelProperties = typeName === "Composite" && (bodyMapper as CompositeMapper).type.modelProperties || {};
     const isPageableResponse = Object.keys(modelProperties).some(k => modelProperties[k].serializedName === "");
     if (typeName === "Sequence" || isPageableResponse) {
-      const arrayResponse = [...(_response.parsedBody || [])] as RestResponse & any[];
+      // We're expecting a sequece(array) make sure that the response body is in the
+      // correct format, if not make it an empty array []
+      const parsedBody =
+        _response.parsedBody && Array.isArray(_response.parsedBody)
```

Comment:
Array.isArray() on undefined or null returns false, so we can skip the check on parsedBody being falsy here

```suggestion
        Array.isArray(_response.parsedBody)
```